pam/internal/adapter/utils.go

```diff
@@ -65,6 +73,46 @@ func isSSHSession(mTx pam.ModuleTransaction) bool {
 	return isSSHSessionValue
 }
 
+func getPamTty(mTx pam.ModuleTransaction) (*os.File, func(), error) {
+	pamTty, err := mTx.GetItem(pam.Tty)
+	if err != nil {
+		return nil, func() {}, err
+	}
+
+	if pamTty == "" {
+		return nil, func() {}, nil
+	}
+
+	tty, err := os.OpenFile(pamTty, os.O_RDWR, 0600)
+	if err != nil {
+		return nil, func() {}, err
+	}
+	cleanup := func() { tty.Close() }
+
+	// We check the fd could be passed to x/term to decide if we should fallback to stdin
+	if tty.Fd() > math.MaxInt {
+		return nil, cleanup, fmt.Errorf("unexpected large PAM TTY fd: %d", tty.Fd())
+	}
+
+	return tty, cleanup, nil
+}
+
+// IsTerminalTTY returns whether the [pam.Tty] or the [os.Stdin] is a terminal TTY.
+func IsTerminalTTY(mTx pam.ModuleTransaction) bool {
+	isTerminalTtyOnce.Do(func() {
+		tty, cleanup, err := getPamTty(mTx)
+		if err != nil {
+			log.Debugf(context.TODO(), "Failed to open PAM TTY: %s", err)
```

Comment:
Shouldn't this be a warning?

Reply:
I didn't want to be too pedantic here, since it's not a big deal on `pam.ErrBadItem` in other failures, but I can probably filter the errors we don't care out...

Reply:
I don't understand. Which errors do we not care about?

Reply:
In the code I see you now explicitly ignore `pam.ErrBadItem`. Why?

Reply:
Oh, I had the bad memory that it was returned when the item wasn't there, but it was wrong and I though I had already removed... But actually I didn't push it.

So it's dropped now.